src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -0,0 +1,284 @@
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPattern, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics();
+            var expectedOutput = @"
+Test
+-1
+Test
+Prop1
+1
+Test
+Prop1
+Prop2
+2
+Test
+Prop1
+Prop2
+-1";
+            var verifier = CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            verifier.VerifyIL("C.Test", @"
+{
+  // Code size       50 (0x32)
+  .maxstack  1
+  .locals init (int V_0,
+                C V_1)
+  IL_0000:  ldstr      ""Test""
+  IL_0005:  call       ""void System.Console.WriteLine(string)""
+  IL_000a:  ldarg.0
+  IL_000b:  brfalse.s  IL_0029
+  IL_000d:  ldarg.0
+  IL_000e:  callvirt   ""C C.Prop1.get""
+  IL_0013:  stloc.1
+  IL_0014:  ldloc.1
+  IL_0015:  brfalse.s  IL_0021
+  IL_0017:  ldloc.1
+  IL_0018:  callvirt   ""C C.Prop2.get""
+  IL_001d:  brfalse.s  IL_0025
+  IL_001f:  br.s       IL_0029
+  IL_0021:  ldc.i4.1
+  IL_0022:  stloc.0
+  IL_0023:  br.s       IL_002b
+  IL_0025:  ldc.i4.2
+  IL_0026:  stloc.0
+  IL_0027:  br.s       IL_002b
+  IL_0029:  ldc.i4.m1
+  IL_002a:  stloc.0
+  IL_002b:  ldloc.0
+  IL_002c:  call       ""void System.Console.WriteLine(int)""
+  IL_0031:  ret
+}");
+        }
+
+        [Fact]
+        public void ExtendedPropertyPatterns_04()
+        {
+            var program = @"
+class C
+{
+    public static void Main()
+    {        
+        _ = new C() is { Prop1<int>.Prop2: {} };
+        _ = new C() is { Prop1->Prop2: {} };
+        _ = new C() is { Prop1?.Prop2: {} };
+        _ = new C() is { Prop1[0]: {} };
+    }
+}
+";
+            var compilation = CreateCompilation(program, parseOptions: TestOptions.RegularWithExtendedPropertyPattern, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics(
+                    // (6,26): error CS9000: Identifier or a simple member access expected.
+                    //         _ = new C() is { Prop1<int>.Prop2: {} };
+                    Diagnostic(ErrorCode.ERR_InvalidNameInSubpattern, "Prop1<int>").WithLocation(6, 26),
+                    // (7,26): error CS9000: Identifier or a simple member access expected.
+                    //         _ = new C() is { Prop1->Prop2: {} };
+                    Diagnostic(ErrorCode.ERR_InvalidNameInSubpattern, "Prop1->Prop2").WithLocation(7, 26),
+                    // (8,26): error CS9001: Conditional access may not be used in property subpatterns. Use a simple member access instead.
+                    //         _ = new C() is { Prop1?.Prop2: {} };
+                    Diagnostic(ErrorCode.ERR_ConditionalAccessInSubpattern, "Prop1?.Prop2").WithLocation(8, 26),
+                    // (9,26): error CS9000: Identifier or a simple member access expected.
+                    //         _ = new C() is { Prop1[0]: {} };
+                    Diagnostic(ErrorCode.ERR_InvalidNameInSubpattern, "Prop1[0]").WithLocation(9, 26));
```

Comment:
I think this error message is clear enough, we don't really need ERR_ConditionalAccessInSubpattern.